File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.py
class Solution(object):
    def lengthOfLongestSubstring(self, s):
        unique_stack=[]
        unique=''
        #sodium-mya

        for i in s:
            if i not in unique:
                unique+=i
            else:
                unique_stack.append(len(unique))
                start=unique.index(i)+1
                unique=unique[start:]
                unique+=i
                unique_stack.append(len(unique))
        unique_stack.append(len(unique))
        if not unique_stack:
            return len(s)
        else:
            return max(unique_stack)

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        vector <int> lastIndex(128,-1);
        int l=0, n=s.size(), maxlen=0;
        for (int r=0; r<n; r++){
            if (lastIndex[s[r]] >= l){
                l = lastIndex[s[r]] + 1;
            }
            lastIndex[s[r]] = r;
            maxlen = max (maxlen, r-l+1);
        }
        return maxlen;
    }
};